enterprise_gateway/enterprisegatewayapp.py

```diff
@@ -133,9 +134,26 @@ def init_configurables(self):
             config=self.config,  # required to get command-line options visible
         )
 
-        # Attempt to start persisted sessions
-        # Commented as part of https://github.com/jupyter-server/enterprise_gateway/pull/737#issuecomment-567598751
-        # self.kernel_session_manager.start_sessions()
+        # For B/C purposes, check if session persistence is enabled.  If so, and availability
+        # mode is not enabled, go ahead and default availability mode to 'active-active'.
+        if self.kernel_session_manager.enable_persistence and self.availability_mode is None:
+            self.availability_mode = "active-active"
+            self.log.info(
+                f"Kernel session persistence is enabled but availability mode is not.  "
+                f"Setting EnterpriseGatewayApp.availability_mode to '{self.availability_mode}'."
+            )
+
+        if self.availability_mode is not None:
+            if self.kernel_session_manager.enable_persistence is False:
+                raise TraitError(
+                    f"Availability mode is configured as '{self.availability_mode}', "
+                    f"yet kernel session persistence has not been enabled.  Configure "
+                    f"KernelSessionManager.enable_persistence and restart Enterprise Gateway."
+                )
+
+            # If we're using active-passive availability, attempt to start persisted sessions
+            if self.availability_mode == "active-passive":
+                self.kernel_session_manager.start_sessions()
```

Comment:
@kevin-bates : I went over the previous comments on this decision, but I am still  do not understand completely on "why we wouldn't want to  load all the sessions from persistence at server start" irrespective of the availability_mode?

Reply:
Yeah, I've had similar discussions with myself. :smile:  (It's good to have someone else to talk with about this!)
  
I think the primary issue here is **affinity** and if we load the sessions in _active-active_ then all EG nodes will have a `KernelManager` _thinking_ they are managing the kernel.  In _active-active_, a "second" node will only manage a "previously managed" kernel when the "previously-managing node" has gone down, so there's still only one node managing the kernel (because we always require "node affinity".

Perhaps the terms _active-active_ and _active-passive_ to describe these modes are not quite correct.  As @dnwe pointed out in the issue, they use _active-passive_ as more of a form of _resilency_ than HA (I suppose more for DR).  Perhaps we could spin _active-active_ to _HA_ and _active-passive_ to _DR_?

Thoughts?

Reply:
my thoughts on the naming the availability modes: 
active-passive  -> single_instance
"active-active" -> multi_instance

Reply:
I've updated the values to use the _instance_ references.  Note that I also added code to auto-enable kernel session persistence if not set when availability mode is set.  It felt a little overbearing to require the persistence setting when it's required to use "availability".  So, rather than throw an exception, we'll log an informational message.

Reply:
@kevin-bates : I was going over some other service documentation where I found 2 new terms used to describe the similar availability scenarios:
1. active-passive -> `standalone`
2. active-active -> `replication`

Reply:
Hmm - I think I like these names over "single-instance" and "multi-instance", especially if there's precedent.

@lresende - you just approved this PR.  Are you okay with going with the names "Standalone" and "Replication"?